From 191824c917ea516944fc600884c06ce3106d7ce3 Mon Sep 17 00:00:00 2001
From: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
Date: Sun, 16 Jan 2011 20:31:45 +0100
Subject: [PATCH] xul2: use JS_NewCompartmentAndGlobalObject

Use JS_NewCompartmentAndGlobalObject rather than JS_NewGlobalObject
to be sure that cx->compartment is not NULL when initializing context.
It basicaly does the same, but it creates a temporary compartment
before crearing the global object, and then restore the previous one.
This avoids a segfault with xulrunner from hg
---
 gjs/jsapi-util.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gjs/jsapi-util.c b/gjs/jsapi-util.c
index d94614c..b4792af 100644
--- a/gjs/jsapi-util.c
+++ b/gjs/jsapi-util.c
@@ -255,7 +255,7 @@ gjs_init_context_standard (JSContext       *context)
 {
     JSObject *global;
 #ifdef HAVE_MOZJS_2
-    global = JS_NewGlobalObject(context, &global_class);
+    global = JS_NewCompartmentAndGlobalObject(context, &global_class, NULL);
     if (global == NULL)
         return FALSE;
 #else
-- 
1.7.4.rc2.dirty

